Docs/sphinx_documentation/source/AMReX_Profiling_Tools.rst

```diff
@@ -107,59 +118,220 @@ You must at least instrument main(), i.e
+
+4) Designate a sub-region to profile, :cpp:`BL_PROFILE_REGION`:
+----------------------------------------------------------------------------------------
+
+Often, it's helpful to look at a subset of timers separately from the complete profile. For
+example, you may want to view the timing of a specific timestep or isolate everything inside the "Chemistry"
+part of the code. This can be accomplished by designating profile regions. All timers within a
+named region will be included both in the full analysis, as well as in a separate sub-analysis.
+
+Regions are meant to be large contiguous blocks of code, and should be used sparingly and purposefully
+to produce useful profiling reports. As such, the possible region options are purposefully limited.
+
+Scoped Regions
+~~~~~~~~~~~~~~
+
+When using the Tiny Profiler, the only available region macro is the scoped macro. To create a region
+that profiles the `MyFuncs` code block, including all timers in the "Additional Code" regions, add
+macros in the following way:
+
+::
+
+          {
+              BL_PROFILE_REGION("MyFuncs");
+
+              <Additional Code A>
+
+              {
+                 BL_PROFILE("MyFunc0");
+
+                 MyFunc_0(arg);
+              }
+
+              <Additional Code B>
+
+              {
+                 BL_PROFILE("MyFunc1");
+
+                 MyFunc_1(arg);
+                 <Additional Code C>
+              }
+          }
+
+Named Regions
+~~~~~~~~~~~~~~
+
+If using the Full Profiler, named region objects are also available. These use slightly modified
+``_VAR_``, ``_START_`` and ``_STOP_`` formatting. To include all timers in a region, except the timers
+inside :cpp:`MyFunc0` and :cpp:`MyFunc1`, arrange the macros as follows:
+
+::
+
+          {
+              BL_PROFILE_REGION("MyFuncs()", myfuncs);
+              <Code Block A>
+              BL_PROFILE_REGION_STOP(myfuncs);
+
+              {
+                   MyFunc_0(arg);
+              }
+
+              BL_PROFILE_REGION_START(myfuncs);
+              <Code Block B>
+              BL_PROFILE_REGION_STOP(myfuncs);
+
+              {
+                   MyFunc_1(arg);
+
+                 BL_PROFILE_REGION_START(myfuncs);
+                 <Code Block C>
+                 BL_PROFILE_REGION_STOP(myfuncs);
+              }
+          }
+
+Currently, these macros are only available in the Full Profiler. This was done to control the size of
+the Tiny Profiler output and to allow instrumentation strategies without needing to re-code each time.
+
+
+.. admonition:: A Combined Tiny-Full Region Profiling Strategy
+
+  We recommend using scoped regions in a few select places that will be timed when either the
+  Tiny or Full Profiler is activated. Because named regions will be isolated by the Full Profiler
+  and ignored by the Tiny Profiler, they can be used for regions that you wish to instrument only
+  when the Full Profiler is active.
+
+.. HACK verify intended meaning.
```

Comment:
```suggestion
```
Don't forget to remove this line before merging.

Reply:
Removed this in a later version.